src/components/Table/useColumns.ts

```diff
@@ -1,10 +1,12 @@
-import { Dispatch, SetStateAction, useState } from "react";
+import { useSessionStorage } from "src/hooks/useSessionStorage";
 import { GridColumn, Kinded } from "./GridTable";
 
 export function useColumns<R extends Kinded, S = {}>(
+  storageKey: string,
   tableColumns: GridColumn<R, S>[],
-): [GridColumn<R, S>[], Dispatch<SetStateAction<GridColumn<R, S>[]>>] {
```

Comment:
**suggestion:** Can we make the storage key optional? (It'd need to move to the second parameter)

Reply:
I like ^ (inferring storage key) but fwiw `c.name` isn't required so we'd just need to handle that case.